Parser/Node.cs

```diff
@@ -44,12 +55,43 @@ public record NWriteStmt (bool NewLine, NExpr[] Exprs) : NStmt {
    public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
 }
 
+// A Read statement
+public record NReadStmt (Token Name, Token[] Names) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
 // An assignment statement
 public record NAssignStmt (Token Name, NExpr Expr) : NStmt {
    public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
 }
+
+// A Call statement
+public record NCallStmt (Token Name, NExpr[] Exprs) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// An IF statement
+public record NIfStmt (NExpr Expr, NStmt[] Stmts) : NStmt {

```

Comment:
Change this to
`public record NIfStmt (NExpr Condition NStmt IfPart NStmt? ElsePart)`